### PR TITLE
Use TextEditorElement for VimCommandModeInputView

### PR DESCRIPTION
**Pull request by tmm1**
fixes a bug where you're unable to see the text you're typing when hitting `/` to search

searching is working on this PR, but I introduced a regression into Motion.Till where the search input box shows up whereas previously it was invisible

cc @maxbrunsfeld

### CODE REVIEW

File: lib/view-models/vim-command-mode-input-view.coffee [hunk 1]
```diff
@@ -4,16 +4,21 @@ module.exports =
...
-        @subview 'editor', new TextEditorView(mini: true)
+      @div class: 'editor-container', outlet: 'editorContainer'
 
   initialize: (@viewModel, opts = {})->
     if opts.class?
       @editorContainer.addClass opts.class
 
     if opts.hidden
+      # TODO: this no longer works with atom-text-editor
       @editorContainer.addClass 'hidden-input'
```
Comment (tmm1) [inline lib/view-models/vim-command-mode-input-view.coffee]: The `hidden-input` class was a special case in the deprecated TextEditorView. I'm not sure what the equivalent is for TextEditorElement

**Comment by MattKunze** [comment]
fwiw I have this updated in #545

**Comment by maxbrunsfeld** [comment]
@MattKunze the nice thing about this approach is that it uses the newer `TextEditorElement` class, which avoids having to use `atom-space-pen-views`. That module will work, but it depends on `space-pen` and jQuery, so it'd be nice to use `TextEditorElement` directly instead. It just means we need to do a little more work to update the tests to not use the jQuery interface.

@tmm1 did you still want to work on this? If not, I can pick it up here. I don't think the `hidden-input` thing needs to change. That class is being applied to a `div` that *contains* the text editor element, not the element itself, so I would think it would continue to work.

**Comment by MattKunze** [comment]
It didn't seem like the optimal approach, I was just following the bouncing balls of documentation regarding upgrading the internal views.

**Comment by tmm1** [comment]
Feel free to pick this up. I noticed the old deprecated class had some special handling for hidden-input, and I wasn't able to make the new view hidden.

**Comment by MattKunze** [comment]
I think the motion regression is with handling single-character inputs. The input used to show up, but it was closed after the first character that was typed. That's what hung me up on the other PR was how to handle that with the updated internals.

**Comment by tmm1** [comment]
> The input used to show up, but it was closed after the first character that was typed.

In my testing it never used to show up. There's code in the plugin to explicitly hide it in the singleChar case.

On this PR, it does show up and it goes away as soon as you type a character. But it's awkward to see it pop up briefly and definitely distracting.

**Comment by maxbrunsfeld** [comment]
Fixes #544

File: lib/view-models/vim-command-mode-input-view.coffee [hunk 2]
```diff
@@ -24,35 +29,29 @@ class VimCommandModeInputView extends View
...
-  stopHandlingEvents: ->
-    if @singleChar?
-      @editor.find('input').off 'textInput', @autosubmit
-    @editor.off 'core:confirm', @confirm
-    @editor.off 'core:cancel', @cancel
-    @editor.find('input').off 'blur', @cancel
-
-  autosubmit: (event) =>
-    @editor.setText(event.originalEvent.data)
+      @editorElement.addEventListener('textInput', @autoSubmit)
```
Comment (tmm1) [inline lib/view-models/vim-command-mode-input-view.coffee]: This doesn't seem to work with addEventListener. Maybe the event needs to be keyup or something different

Comment (maxbrunsfeld) [inline lib/view-models/vim-command-mode-input-view.coffee]: Yeah, I can't remember why that doesn't work. Observing text changes on the model seems to work ok though.

**Comment by tmm1** [comment]
Thanks @maxbrunsfeld, this is working great now!

**Pull request merged**